### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by brayann** [twilio/twilio-php#443]
**Version:5.7.1**

### Code Snippet
```php
/**
	 * Moves the user to a new call from another call, and then ends it
	 *
	 * @param Companiesleads $newCall
	 * @param Companiesleads $originalCall
	 * @return mixed
	 */
	public function answerAndHangup($newCall, $originalCall)
	{
		try {
			$this->client // moves the user to the conference of the new call
						->calls($originalCall->calltype->outbound ? $originalCall->CallUUID : $originalCall->ChildCallSid)
						->update(['Url'    => \URL::to("/twilio/conference?CallUUID={$newCall->CallUUID}&transferred=1")]);
			sleep(0.5);
			$this->client // ends the previous call (that he was currently active)
						->calls($originalCall->calltype->outbound ? $originalCall->ChildCallSid : $originalCall->CallUUID)
						->update(['status'    => 'completed']);
			$conference = $this->client->conferences->read(array("friendlyName" => $newCall->CallUUID)); // Here we move the other line to not be on hold, when user enters that conference room
			$conferenceUUID = $conference[0]->sid;
			$this->client
						->conferences($conferenceUUID)
						->participants($newCall->calltype->outbound ? $newCall->ChildCallSid : $newCall->CallUUID)
						->update(['Hold' => 'false']);
			return true;
		} catch (\Exception $e) {
			\Bugsnag::notifyException($e);
			return $e->getMessage();
		}
	}
```

### Exception/Log
```
Symfony\Component\Debug\Exception\FatalErrorException: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes)

Location
----------------
vendor/twilio/sdk/Twilio/Rest/Api/V2010/Account/ConferenceInstance.php:47 - Illuminate\Exception\Handler::handleShutdown

----

Error
----------------
Symfony\Component\Debug\Exception\FatalErrorException: Allowed memory size of 536870912 bytes exhausted (tried to allocate 32768 bytes)

Location
----------------
vendor/twilio/sdk/Twilio/Http/CurlClient.php:36 - Illuminate\Exception\Handler::handleShutdown
```

### Steps to Reproduce
Not really sure but it seems that if you try to do more than a couple of operations on the library, it uses a lot of memory, causing this error to occur.

**Comment by jmctwilio** [comment]
@brayann Thank you for bringing this up.  I'll create a ticket to investigate the cause of this error and fix it.

**Comment by brayann** [comment]
Thanks, please update me if you find something. I had to increase memory with `ini_set('memory_limit')` to bypass this issue.

**Comment by brayann** [comment]
@jmctwilio any news on this? 

Further information:

I checked on NewRelic and saw that Twilio API is being called A LOT from that code that I pasted. Is there anything on the library that keeps calling the API on a loop? 
<img width="742" alt="screen shot 2017-08-14 at 6 49 28 pm" src="https://user-images.githubusercontent.com/253673/29294886-e7ccbc02-8127-11e7-9793-49b7f0dda89c.png">
This is not normal since on my method I only call the API 3 times, so there's something really weird there.
